src/combinatorics/simple_graph/basic.lean

```diff
@@ -317,6 +317,30 @@ instance decidable_mem_edge_set [decidable_rel G.adj] :
 instance edges_fintype [decidable_eq V] [fintype V] [decidable_rel G.adj] :
   fintype G.edge_set := subtype.fintype _
 
+section from_edge_set
+
+variable (s : set (sym2 V))
+
+/--
+`from_edge_set` constructs a `simple_graph` from a set of edges, without loops.
+-/
+def from_edge_set : simple_graph V :=
+{ adj := λ v w, v ≠ w ∧ ⟦(v, w)⟧ ∈ s,
+  symm := λ v w h, ⟨h.1.symm, by {rw sym2.eq_swap, exact h.2}⟩ }
+
+@[simp] lemma adj_from_edge_set : (from_edge_set s).adj v w ↔ v ≠ w ∧ ⟦(v, w)⟧ ∈ s := iff.rfl
+
+@[simp] lemma from_edge_set_eq : from_edge_set (G.edge_set) = G :=
```

Comment:
Perhaps this should get the somewhat silly name `from_edge_set_edge_set` so that it has more parallelism with the following two lemmas.